conversation_reconstruction_dataflow/ingest_revisions/dataflow_main.py

```diff
@@ -68,19 +71,36 @@ def run(known_args, pipeline_args):
   with beam.Pipeline(options=pipeline_options) as p:
     pcoll = (p | ReadFromText(known_args.input)
                    | beam.ParDo(WriteDecompressedFile())
-                   | beam.io.Write(bigquery.BigQuerySink(known_args.table, schema=known_args.schema)))
+                   | beam.io.Write(bigquery.BigQuerySink(known_args.table, schema=known_args.schema, validate = True)))
 
 def truncate_content(s):
     dic = json.loads(s) 
     dic['truncated'] = False
-    if sys.getsizeof(s) > THERESHOLD:
+    dic['no_records'] = 1
+    dic['record_index'] = 0
+    filesize = sys.getsizeof(s)
+    if filesize > THERESHOLD:
+       pieces = math.ceil(filesize / THERESHOLD)
```

Comment:
You'll get simpler logic and behavior if, instead of trying to divide the size up equally, you simply put the first, up to, THRESHOLD chunk into the first dic, then the second up to THRESHOLD size into the next one, and so on.

Reply:
Hmm THERESHOLD is the thereshold of allocated memory size of the string, and I divided the string using length of the string. Frequently checking the allocated memory size might be a bit inefficient?